sdk/python/kfp/onprem.py

```diff
@@ -11,26 +16,35 @@ def mount_pvc(pvc_name='pipeline-claim', volume_name='pipeline', volume_mount_pa
             train = train_op(...)
             train.apply(mount_pvc('claim-name', 'pipeline', '/mnt/pipeline'))
     """
+
     def _mount_pvc(task):
         from kubernetes import client as k8s_client
         # there can be other ops in a pipeline (e.g. ResourceOp, VolumeOp)
         # refer to #3906
-        if not hasattr(task, "add_volume") or not hasattr(task, "add_volume_mount"):
+        if not hasattr(task, "add_volume") or not hasattr(task,
+                                                          "add_volume_mount"):
             return task
-        local_pvc = k8s_client.V1PersistentVolumeClaimVolumeSource(claim_name=pvc_name)
+        local_pvc = k8s_client.V1PersistentVolumeClaimVolumeSource(
+            claim_name=pvc_name
+        )
         return (
-            task
-                .add_volume(
-                    k8s_client.V1Volume(name=volume_name, persistent_volume_claim=local_pvc)
+            task.add_volume(
+                k8s_client.V1Volume(
+                    name=volume_name, persistent_volume_claim=local_pvc
                 )
-                .add_volume_mount(
-                    k8s_client.V1VolumeMount(mount_path=volume_mount_path, name=volume_name)
+            ).add_volume_mount(
+                k8s_client.V1VolumeMount(
+                    mount_path=volume_mount_path, name=volume_name
                 )
+            )
         )
+
     return _mount_pvc
 
 
-def use_k8s_secret(secret_name: str = 'k8s-secret', k8s_secret_key_to_env: Dict = {}):
+def use_k8s_secret(
+    secret_name: str = 'k8s-secret', k8s_secret_key_to_env: Dict = {}
```

Comment:
Since you're touching this line, this should be `Optional[dict] = None` and inside the function body, defaulting to `{}` if it's `None`.

We should avoid mutable default values.  See: https://github.com/kubeflow/pipelines/pull/6099#issue-694188856

Reply:
Good catch!